src/LessCompiler.ts

```diff
@@ -68,9 +68,12 @@ export async function compile(
 
   options.plugins.push(new LessDocumentResolverPlugin());
 
-  // If options.rootFileInfo is not undefined, less will use the filepath to read content again
-  if (useContent) {
+  if (preprocessors.length > 0) {
+    // If options.rootFileInfo is not undefined, less will use the filepath to read content again.
     delete options.rootFileInfo;
+    // Used to cache some variables for use by other preprocessors.
+    const ctx = new Map<string, any>();
+    content = await preprocessors.reduce(async (result, p) => p(await result, ctx), Promise.resolve(content));
```

Comment:
This is pretty amazing! I was not aware you can use `reduce` with promises.

For the sake of readability, however, I'd probably prefer we use a simple `for` loop:
```typescript
for await (const p of preprocessors) {
  content = await p(result, ctx);
}
```

I believe this is easier for the reader to comprehend. I'll accept the PR as is, and then amend this before I release. No change required.